Nu/Nu/Ecs/EcsPrelude.fs

```diff
@@ -45,32 +46,50 @@ type Store =
         end
 
 /// Stores components for an Ecs.
-type Store<'c when 'c : struct and 'c :> 'c Component> (name) =
+type Store<'c when 'c: struct and 'c :> 'c Component>(name) =
     let mutable arr = Array.zeroCreate<'c> Constants.Ecs.ArrayReserve
     member this.Length = arr.Length
     member this.Name = name
     member this.Item i = &arr.[i]
     member this.SetItem index comp = arr.[index] <- comp
     member this.ZeroItem index = arr.[index] <- Unchecked.defaultof<'c>
-    member this.Grow () =
+
+    member this.Grow() =
         let length = int (single (max arr.Length 2) * 1.5f)
         let arr' = Array.zeroCreate<'c> length
-        Array.Copy (arr, arr', arr.Length)
+        Array.Copy(arr, arr', arr.Length)
         arr <- arr'
-    member this.Read index count (stream : FileStream) =
+
+    member this.Write(stream: FileStream) =
+        let arr = Branchless.reinterpret arr
+        for i = 0 to  this.Length * sizeof<'c> - 1 do
+            let b = Unsafe.ReadUnaligned(&Unsafe.Add(&MemoryMarshal.GetArrayDataReference arr, i )) 
+            stream.WriteByte b
+
+        stream.Flush()
+        stream.Close()
+
+    member this.Read index count (stream: FileStream) =
         let compSize = sizeof<'c>
         let comp = Unchecked.defaultof<'c> :> obj
         let buffer = Array.zeroCreate<byte> compSize
-        let gch = GCHandle.Alloc (comp, GCHandleType.Pinned)
-        try 
+        let gch = GCHandle.Alloc(comp, GCHandleType.Pinned)
+
+        try
             let mutable index = index
+
             for _ in 0 .. dec count do
-                stream.Read (buffer, 0, compSize) |> ignore<int>
-                Marshal.Copy (buffer, 0, gch.AddrOfPinnedObject (), compSize)
-                if index = arr.Length then this.Grow ()
+                stream.Read(buffer, 0, compSize) |> ignore<int>
```

Comment:
We should probably also verify that stream.Read was successful, throwing an exception if it didn't read all the requested byte.